### PR TITLE
backup: make LogBackupTemplate optional

### PR DESCRIPTION
**Pull request by Ehco1996**
### What problem does this PR solve?

close https://github.com/pingcap/tidb-operator/issues/4955
### What is changed and how does it work?

as title

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
none
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -2200,6 +2200,7 @@ type BackupScheduleSpec struct {
 	// BackupTemplate is the specification of the backup structure to get scheduled.
 	BackupTemplate BackupSpec `json:"backupTemplate"`
 	// LogBackupTemplate is the specification of the log backup structure to get scheduled.
+	// +optional
```
Comment (Ehco1996) [inline pkg/apis/pingcap/v1alpha1/types.go:2203]: only one line change, this field was already optional during implementation, it's just that the comment wasn't written correctly when generating the CRD.
see more in  https://github.com/pingcap/tidb-operator/blob/e2eb643abb916684c1dfcb420441561300ddf2c0/pkg/backup/backupschedule/backup_schedule_manager.go#L138

Comment (Ehco1996) [reply pkg/apis/pingcap/v1alpha1/types.go:2203]: ci failed when checking forward compatibility, maybe we should force merge this pr.

https://github.com/pingcap/tidb-operator/actions/runs/4605029707/jobs/8136576127?pr=4956

**Review by grovecai — APPROVED** [review]

**Comment by Ehco1996** [comment]
br it test passed https://ci.pingcap.net/blue/organizations/jenkins/tidb-operator-pull-e2e-kind-br/detail/tidb-operator-pull-e2e-kind-br/2604/pipeline

**Pull request merged**